pkg/component/logging/vali/vali.go

```diff
@@ -0,0 +1,1104 @@
+	valiUserAndGroupId      = 10001
+	valiConfigMapVolumeName = "config"
+	valiPVCName             = "vali"
+)
+
+var (
+	//go:embed templates/curator-config.yaml
+	curatorConfigBytes []byte
+	//go:embed templates/vali-init.sh
+	valiInitBytes []byte
+
+	tplNameValiConfig = "vali-config"
+	//go:embed templates/vali-config.tpl
+	tplContentValiConfig string
+	valiConfigTmpl       *template.Template
+	//valiConfigYAMLTmpl     = template.Must(template.ParseFS(valiConfigYAMLTmplFile, "templates/vali-config.tpl"))
+
+	tplNameTelegrafConfig = "telegraf-config"
+	//go:embed templates/telegraf-config.tpl
+	tplContentTelegrafConfig string
+	telegrafConfigTmpl       *template.Template
+	//telegrafConfigYAMLTmpl   = template.Must(template.ParseFS(telegrafConfigYAMLTmplFile, "telegraf-config.tpl"))
+
+	tplNameTelegrafStart = "telegraf-start"
+	//go:embed templates/telegraf-start.sh.tpl
+	tplContentTelegrafStart string
+	telegrafStartTmpl       *template.Template
+	//telegrafStartTmpl     = template.Must(template.ParseFS(telegrafStartTmplFile, "templates/telegraf-start.sh.tpl"))
+
+	controlledValues            = vpaautoscalingv1.ContainerControlledValuesRequestsOnly
+	containerPolicyOff          = vpaautoscalingv1.ContainerScalingModeOff
+	fsGroupChangeOnRootMismatch = corev1.FSGroupChangeOnRootMismatch
+	pathType                    = networkingv1.PathTypePrefix
+
+	ingressTLSCertificateValidity = 730 * 24 * time.Hour // ~2 years, see https://support.apple.com/en-us/HT210176
+)
+
+func init() {
+	var err error
+	valiConfigTmpl, err = template.
+		New(tplNameValiConfig).
+		Funcs(sprig.TxtFuncMap()).
+		Parse(tplContentValiConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	telegrafConfigTmpl, err = template.
+		New(tplNameTelegrafStart).
+		Funcs(sprig.TxtFuncMap()).
+		Parse(tplContentTelegrafConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	telegrafStartTmpl, err = template.
+		New(tplNameTelegrafConfig).
+		Funcs(sprig.TxtFuncMap()).
+		Parse(tplContentTelegrafStart)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Values are the values for the Vali.
+type Values struct {
+	Replicas              int32
+	AuthEnabled           bool
+	Storage               *resource.Quantity
+	HvpaEnabled           bool
+	MaintenanceBegin      string
+	MaintenanceEnd        string
+	ValiImage             string
+	CuratorImage          string
+	RenameLokiToValiImage string
+	InitLargeDirImage     string
+	PriorityClassName     string
+	TelegrafImage         string
+	KubeRBACProxyImage    string
+	ClusterType           string
+	IngressClass          string
+	ValiHost              string
+}
+
+type vali struct {
+	client         client.Client
+	namespace      string
+	secretsManager secretsmanager.Interface
+	k8sVersion     *semver.Version
+	values         Values
+}
+
+var _ component.Deployer = &vali{}
+
+// New creates a new instance of Vali deployer.
+func New(
+	client client.Client,
+	namespace string,
+	secretsManager secretsmanager.Interface,
+	k8sVersion *semver.Version,
```

Comment:
Which version is this? The version of the runtime cluster? If so, call it `runtimeKubernetesVersion`